### PR TITLE
add api set feed setting

### PR DESCRIPTION
**Pull request by NNagato**

### CODE REVIEW

File: data/storage/bolt.go [hunk 1]
```diff
@@ -287,6 +291,117 @@ func (bs *BoltStorage) GetFeedConfiguration() ([]common.FeedConfiguration, error
 	return results, err
 }
 
+func (bs *BoltStorage) StorePendingFeedSetting(value []byte) error {
+	var (
+		err         error
+		pendingData common.MapFeedSetting
```
Comment (secmask) [inline data/storage/bolt.go]: we should verify that each feed in update request exist in AllFeed() list.

File: common/feeds.go [hunk 2]
```diff
@@ -4,3 +4,9 @@ type FeedConfiguration struct {
 	Name    string `json:"name"`
 	Enabled bool   `json:"enabled"`
 }
+
+type FeedSetting struct {
```
Comment (secmask) [inline common/feeds.go]: public type should have comment.

File: common/feeds.go [hunk 3]
```diff
@@ -4,3 +4,9 @@ type FeedConfiguration struct {
...
+	BaseVolatilitySpread float64 `json:"base_volatility_spread"`
+}
+
+type MapFeedSetting map[string]FeedSetting
```
Comment (secmask) [inline common/feeds.go]: public type should have comment.

File: common/feeds.go (hunk 2 above)
Comment (NNagato) [reply common/feeds.go]: fixed in https://github.com/KyberNetwork/reserve-data/pull/787/commits/7ee2f70c0bdd6a8b4aba5e17c9adc44560af6890

File: common/feeds.go (hunk 3 above)
Comment (NNagato) [reply common/feeds.go]: fixed in https://github.com/KyberNetwork/reserve-data/pull/787/commits/7ee2f70c0bdd6a8b4aba5e17c9adc44560af6890

File: data/storage/bolt.go (hunk 1 above)
Comment (NNagato) [reply data/storage/bolt.go]: fixed in https://github.com/KyberNetwork/reserve-data/pull/787/commits/7ee2f70c0bdd6a8b4aba5e17c9adc44560af6890

**Comment by NNagato** [comment]
> we should extract inner loop into function, so we don't have to check `if index == len(allFeeds)-1 {`

pls take a look, I updated in https://github.com/KyberNetwork/reserve-data/pull/787/commits/32befd4fa120f6b8ca4e1c4b0fbec89e321cd6f9

**Pull request merged**